In [556]:
%matplotlib inline
import matplotlib
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import cv2
from sklearn import metrics
import seaborn as sns
import sklearn
import math
from sklearn.linear_model import Ridge
from sklearn.linear_model import Lasso
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import OneHotEncoder

In [557]:
df=pd.read_csv('problem_2_train.csv')
dft=pd.read_csv('problem_2_test.csv',index_col=0)
df.head()

srno mobmake  hubid  hubpincode  pickuppartnerid  deliverypartnerid  \
0  2851218      AN      7      500004              233                233   
1  2838330      AJ     10      226010              233                233   
2  2835781      AJ      2      110018              233                233   
3  2838589      AM      2      110018              233                233   
4  2855214      AN     21      110015              233                233   

   insuranceCompanyId custpincodecategory claimtype  custpincode  \
0                 131                   A      ADLD       500001   
1                 131                   A      ADLD       226022   
2                 228                   A      ADLD       110018   
3                 228                   B      ADLD       124001   
4                 131                   A      ADLD       201301   

  memcreateddate mobilepurchasedate claimintdate  servicecntrid  \
0     2017-04-27         2016-12-02   2018-02-08           4403   
1     2017-03-06         2017-03-05   2018-02-04           4444   
2     2018-02-01         2018-01-27   2018-02-03           4388   
3     2018-01-17         2018-01-31   2018-02-04           4092   
4     2018-01-23         2018-01-20   2018-02-09           4580   

  pickupStartdate  tat_in_days  
0      2018-02-13            5  
1      2018-02-08           12  
2      2018-02-12            9  
3      2018-02-14            9  
4      2018-02-14            8

In [558]:
dft.head()

srno  mobmake  hubid  hubpincode  pickuppartnerid  deliverypartnerid  \
2  2894489       10      3      560011              233                233   
3  2876708       10      2      110018              233                233   
4  2890283       13     22      400104              233                233   
5  2897438       15      6      160059              233                233   
8  2941799       13     21      110015              233                233   

   insuranceCompanyId  custpincodecategory  claimtype  custpincode  \
2                 228                    2          0       590014   
3                 228                    1          0       250110   
4                 131                    2          0       496001   
5                 131                    1          0       160036   
8                 131                    1          0       124001   

  memcreateddate mobilepurchasedate claimintdate  servicecntrid  \
2     2018-01-12         2018-01-11   2018-02-21           4460   
3     2017-08-07         2017-05-21   2018-02-16           4388   
4     2017-11-10         2017-09-22   2018-02-20           4581   
5     2017-12-12         2017-12-10   2018-02-22           4549   
8     2017-12-31         2017-06-26   2018-03-05           4580   

  pickupStartdate  pickupStartdate_Code    claimintdate_Code  tat_in_days  
2      2018-02-28   1519776000000000000  1519171200000000000           13  
3      2018-02-26   1519603200000000000  1518739200000000000           13  
4      2018-02-23   1519344000000000000  1519084800000000000           13  
5      2018-02-27   1519689600000000000  1519257600000000000           11  
8      2018-03-14   1520985600000000000  1520208000000000000           11

In [559]:
df.claimtype.unique()

array(['ADLD', 'EW'], dtype=object)

### Converting categorical data(mobmake,claimtype,custpincodecategory) of train file into numbers.

In [560]:
le=LabelEncoder()
dfle=df
dfle.mobmake=le.fit_transform(dfle.mobmake)

In [561]:
le=LabelEncoder()
dfle=df
dfle.claimtype=le.fit_transform(dfle.claimtype)

In [562]:
le=LabelEncoder()
dfle=df
dfle.custpincodecategory=le.fit_transform(dfle.custpincodecategory)

In [563]:
df.head()

srno  mobmake  hubid  hubpincode  pickuppartnerid  deliverypartnerid  \
0  2851218       12      7      500004              233                233   
1  2838330        8     10      226010              233                233   
2  2835781        8      2      110018              233                233   
3  2838589       11      2      110018              233                233   
4  2855214       12     21      110015              233                233   

   insuranceCompanyId  custpincodecategory  claimtype  custpincode  \
0                 131                    1          0       500001   
1                 131                    1          0       226022   
2                 228                    1          0       110018   
3                 228                    2          0       124001   
4                 131                    1          0       201301   

  memcreateddate mobilepurchasedate claimintdate  servicecntrid  \
0     2017-04-27         2016-12-02   2018-02-08           4403   
1     2017-03-06         2017-03-05   2018-02-04           4444   
2     2018-02-01         2018-01-27   2018-02-03           4388   
3     2018-01-17         2018-01-31   2018-02-04           4092   
4     2018-01-23         2018-01-20   2018-02-09           4580   

  pickupStartdate  tat_in_days  
0      2018-02-13            5  
1      2018-02-08           12  
2      2018-02-12            9  
3      2018-02-14            9  
4      2018-02-14            8

### Converting categorical data(mobmake,claimtype,custpincodecategory) of test file into numbers.

In [564]:
le=LabelEncoder()
dfle=dft
dfle.mobmake=le.fit_transform(dfle.mobmake)

le=LabelEncoder()
dfle=dft
dfle.claimtype=le.fit_transform(dfle.claimtype)

le=LabelEncoder()
dfle=dft
dfle.custpincodecategory=le.fit_transform(dfle.custpincodecategory)

In [565]:
dft.head()

srno  mobmake  hubid  hubpincode  pickuppartnerid  deliverypartnerid  \
2  2894489        7      3      560011              233                233   
3  2876708        7      2      110018              233                233   
4  2890283       10     22      400104              233                233   
5  2897438       11      6      160059              233                233   
8  2941799       10     21      110015              233                233   

   insuranceCompanyId  custpincodecategory  claimtype  custpincode  \
2                 228                    1          0       590014   
3                 228                    0          0       250110   
4                 131                    1          0       496001   
5                 131                    0          0       160036   
8                 131                    0          0       124001   

  memcreateddate mobilepurchasedate claimintdate  servicecntrid  \
2     2018-01-12         2018-01-11   2018-02-21           4460   
3     2017-08-07         2017-05-21   2018-02-16           4388   
4     2017-11-10         2017-09-22   2018-02-20           4581   
5     2017-12-12         2017-12-10   2018-02-22           4549   
8     2017-12-31         2017-06-26   2018-03-05           4580   

  pickupStartdate  pickupStartdate_Code    claimintdate_Code  tat_in_days  
2      2018-02-28   1519776000000000000  1519171200000000000           13  
3      2018-02-26   1519603200000000000  1518739200000000000           13  
4      2018-02-23   1519344000000000000  1519084800000000000           13  
5      2018-02-27   1519689600000000000  1519257600000000000           11  
8      2018-03-14   1520985600000000000  1520208000000000000           11

In [566]:
a=df.custpincodecategory.unique()
a

array([1, 2, 3, 0, 4], dtype=int64)

### Checking for null values in both the files

In [567]:
print(df.custpincodecategory.value_counts())
print()
print(dft.custpincodecategory.value_counts())

1    4020
2    1346
3     554
0     186
4     144
Name: custpincodecategory, dtype: int64

0    475
1    204
2     44
Name: custpincodecategory, dtype: int64


In [568]:
train_null = df.query("custpincodecategory == 0").index
test_null = dft.query("custpincodecategory == 0").index

### Droping null values from both the files

In [569]:
df.drop(train_null,inplace = True)
dft.drop(test_null,inplace = True)

In [570]:
df.custpincodecategory.value_counts()

1    4020
2    1346
3     554
4     144
Name: custpincodecategory, dtype: int64

In [571]:
df.describe().T

count          mean            std        min  \
srno                 6064.0  2.946472e+06   77530.310085  2818359.0   
mobmake              6064.0  1.219030e+01       5.922675        0.0   
hubid                6064.0  1.164726e+01       9.161700        1.0   
hubpincode           6064.0  3.415287e+05  213115.116740   110015.0   
pickuppartnerid      6064.0  2.336268e+02      29.471301        0.0   
deliverypartnerid    6064.0  2.319832e+02      21.454931        0.0   
insuranceCompanyId   6064.0  1.551060e+02      41.922241      131.0   
custpincodecategory  6064.0  1.475923e+00       0.758098        1.0   
claimtype            6064.0  6.596306e-04       0.025677        0.0   
custpincode          6064.0  3.666730e+05  209754.580730   110001.0   
servicecntrid        6064.0  4.456731e+03     201.632269     4008.0   
tat_in_days          6064.0  1.257602e+01       4.947839        2.0   

                           25%        50%         75%        max  
srno                 2877245.0  2944960.5  3013050.75  3101035.0  
mobmake                    8.0       12.0       17.00       37.0  
hubid                      3.0        8.0       21.00       27.0  
hubpincode            110018.0   400059.0   560011.00   800001.0  
pickuppartnerid          233.0      233.0      233.00      308.0  
deliverypartnerid        233.0      233.0      233.00      245.0  
insuranceCompanyId       131.0      131.0      131.00      228.0  
custpincodecategory        1.0        1.0        2.00        4.0  
claimtype                  0.0        0.0        0.00        1.0  
custpincode           134109.0   395009.0   560018.00   855107.0  
servicecntrid           4398.0     4474.0     4580.00     5146.0  
tat_in_days                9.0       12.0       15.00       57.0

In [572]:
df.head(3)

srno  mobmake  hubid  hubpincode  pickuppartnerid  deliverypartnerid  \
0  2851218       12      7      500004              233                233   
1  2838330        8     10      226010              233                233   
2  2835781        8      2      110018              233                233   

   insuranceCompanyId  custpincodecategory  claimtype  custpincode  \
0                 131                    1          0       500001   
1                 131                    1          0       226022   
2                 228                    1          0       110018   

  memcreateddate mobilepurchasedate claimintdate  servicecntrid  \
0     2017-04-27         2016-12-02   2018-02-08           4403   
1     2017-03-06         2017-03-05   2018-02-04           4444   
2     2018-02-01         2018-01-27   2018-02-03           4388   

  pickupStartdate  tat_in_days  
0      2018-02-13            5  
1      2018-02-08           12  
2      2018-02-12            9

In [573]:
a=df.custpincodecategory.unique()
a

array([1, 2, 3, 4], dtype=int64)

In [574]:
df.dtypes

srno                    int64
mobmake                 int32
hubid                   int64
hubpincode              int64
pickuppartnerid         int64
deliverypartnerid       int64
insuranceCompanyId      int64
custpincodecategory     int32
claimtype               int32
custpincode             int64
memcreateddate         object
mobilepurchasedate     object
claimintdate           object
servicecntrid           int64
pickupStartdate        object
tat_in_days             int64
dtype: object

### Changing the datatype of Dates Column(memcreateddate,mobilepurchasedate,claimdate,pickupStartdate) 

In [575]:
df['memcreateddate']=pd.to_datetime(df['memcreateddate'])
df['mobilepurchasedate']=pd.to_datetime(df['mobilepurchasedate'])
df['claimintdate']=pd.to_datetime(df['claimintdate'])
df['pickupStartdate']=pd.to_datetime(df['pickupStartdate'])

dft['memcreateddate']=pd.to_datetime(dft['memcreateddate'])
dft['mobilepurchasedate']=pd.to_datetime(dft['mobilepurchasedate'])
dft['claimintdate']=pd.to_datetime(dft['claimintdate'])
dft['pickupStartdate']=pd.to_datetime(dft['pickupStartdate'])

print(df.dtypes)
print()
print(dft.dtypes)

srno                            int64
mobmake                         int32
hubid                           int64
hubpincode                      int64
pickuppartnerid                 int64
deliverypartnerid               int64
insuranceCompanyId              int64
custpincodecategory             int32
claimtype                       int32
custpincode                     int64
memcreateddate         datetime64[ns]
mobilepurchasedate     datetime64[ns]
claimintdate           datetime64[ns]
servicecntrid                   int64
pickupStartdate        datetime64[ns]
tat_in_days                     int64
dtype: object

srno                             int64
mobmake                          int32
hubid                            int64
hubpincode                       int64
pickuppartnerid                  int64
deliverypartnerid                int64
insuranceCompanyId               int64
custpincodecategory              int32
claimtype                        int32
custpincode               

In [284]:
# plt.scatter(df.mobmake,df.claimtype,df.tat_in_days)

### Making two more columns for in both the files

In [576]:
df['pickupStartdate_Code']=pd.to_numeric(df['pickupStartdate'])
df['claimintdate_Code'] = pd.to_numeric(df['claimintdate'])

dft['pickupStartdate_Code']=pd.to_numeric(dft['pickupStartdate'])
dft['claimintdate_Code'] = pd.to_numeric(dft['claimintdate'])


In [634]:

dft.head()

srno  mobmake  hubid  hubpincode  pickuppartnerid  deliverypartnerid  \
2   2894489        7      3      560011              233                233   
4   2890283       10     22      400104              233                233   
31  2839241        4     17      800001              233                233   
36  2972765       10     22      400104                0                233   
41  3014945       10     26      560066              201                233   

    insuranceCompanyId  custpincodecategory  claimtype  custpincode  \
2                  228                    1          0       590014   
4                  131                    1          0       496001   
31                 131                    1          0       824124   
36                 131                    2          0       382022   
41                 131                    2          0       630606   

   memcreateddate mobilepurchasedate claimintdate  servicecntrid  \
2      2018-01-12         2018-01-11   2018-02-21           4460   
4      2017-11-10         2017-09-22   2018-02-20           4581   
31     2017-02-20         2017-02-17   2018-02-05           4538   
36     2017-08-18         2017-08-17   2018-03-12           4581   
41     2017-12-28         2017-08-01   2018-03-24           5001   

   pickupStartdate  pickupStartdate_Code    claimintdate_Code  tat_in_days  
2       2018-02-28   1519776000000000000  1519171200000000000           13  
4       2018-02-23   1519344000000000000  1519084800000000000           13  
31      2018-02-08   1518048000000000000  1517788800000000000           13  
36      2018-03-16   1521158400000000000  1520812800000000000           13  
41      2018-03-27   1522108800000000000  1521849600000000000           12

### Feature Selection(Selecting all features)

In [635]:
feature_cols=['srno','hubid','hubpincode','pickuppartnerid','deliverypartnerid','insuranceCompanyId','custpincode','servicecntrid']

In [636]:
X=df[feature_cols]
X.shape

(6064, 8)

In [637]:
y=df['tat_in_days']

In [638]:
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=1)

In [639]:
print(X_train.shape)
print(X_test.shape)
print(y_test.shape)
print(y_train.shape)

(4244, 8)
(1820, 8)
(1820,)
(4244,)


In [640]:
reg = LinearRegression()
reg.fit(X_train,y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [641]:
print(reg.intercept_)
print(reg.coef_)

66.23608161396747
[-1.10752976e-05 -2.01343454e-02 -7.83529588e-06 -4.05431456e-03
 -1.31868574e-02 -5.23809696e-03  9.18400864e-06 -3.72821780e-03]


In [642]:
y_pred=reg.predict(X_test)
print(y_test,y_pred)

3308    23
310      7
2503     9
886     18
3133    10
2874    11
243     13
4333    13
5661    10
4197     9
1196     7
4163    17
294     20
1575    12
4712    11
4732     9
1446     9
3950     6
920     11
80      19
2515    11
4436    13
323     14
2673     8
2056    19
3892    16
6110    23
5616    14
3414     9
5384     7
        ..
6199    12
1398    10
4501    17
5371    13
912      9
282     14
6187    15
1770     7
3873    12
3087     6
5934    16
2617     9
1451    22
100     21
4257     9
4410    14
5241    12
346     11
2645     9
1151    13
4843    12
40       7
5669     7
6049    17
3394    14
3487    16
2667    11
4234     7
2729    12
940     11
Name: tat_in_days, Length: 1820, dtype: int64 [12.59629558 11.242533   13.86310292 ... 10.8831389  13.23394883
 11.72778284]


In [643]:
RMSE=np.sqrt(metrics.mean_squared_error(y_test,y_pred))
RMSE

4.688887685949893

In [644]:
reg.score(X,y)

0.08292316869586791

#### RMSE Score is 4.688887685949893

### 1. Changing selection criteria(leaving some columns)

In [645]:
feature_cols=['mobmake','hubpincode','pickuppartnerid','deliverypartnerid','insuranceCompanyId','custpincodecategory','claimtype','custpincode','servicecntrid','pickupStartdate_Code','claimintdate_Code']

In [648]:
X=df[feature_cols]
X.shape

(6064, 11)

In [649]:
y=df['tat_in_days']

In [650]:
X_train, X_test, y_train, y_test = train_test_split(X,y,random_state=1)

In [651]:
print(X_train.shape)
print(X_test.shape)
print(y_train.shape)
print(y_test.shape)


(4548, 11)
(1516, 11)
(4548,)
(1516,)


### Linear Regression

In [652]:
linreg = LinearRegression()
linreg.fit(X_train,y_train)
y_pred = linreg.predict(X_test)
y_pred = y_pred.astype(int)

In [303]:
# y_pred

In [653]:
lr=np.sqrt(metrics.mean_squared_error(y_test,y_pred))
lr

4.462094826140186

### Lasso Regression

In [654]:
lasso = Lasso()
lasso.fit(X_train,y_train)
y_pred = lasso.predict(X_test)
y_pred = y_pred.astype(int)

In [655]:
lar=np.sqrt(metrics.mean_squared_error(y_test,y_pred))
lar

4.632824266688283

### Ridge Regression

In [656]:
ridge1 = Ridge()
ridge1.fit(X_train,y_train)
y_pred = ridge1.predict(X_test)
y_pred = y_pred.astype(int)

c:\users\gunjan\appdata\local\programs\python\python37-32\lib\site-packages\sklearn\linear_model\ridge.py:147: LinAlgWarning: Ill-conditioned matrix (rcond=1.34261e-34): result may not be accurate.
  overwrite_a=True).T


In [657]:
rr=np.sqrt(metrics.mean_squared_error(y_test,y_pred))
rr

4.450771478588562

### Model Evaluation

In [658]:
models = pd.DataFrame({
    'Model': [ 'Linear Regression','Lasso Regression','Ridge Regression'],
    'RMSE Score': [lr,lar,rr]})
models.sort_values(by='RMSE Score', ascending=True)

Model  RMSE Score
2   Ridge Regression    4.450771
0  Linear Regression    4.462095
1   Lasso Regression    4.632824

#### RMSE Score through Ridge Regression=4.450771

In [601]:
df.head()

srno  mobmake  hubid  hubpincode  pickuppartnerid  deliverypartnerid  \
0  2851218       12      7      500004              233                233   
1  2838330        8     10      226010              233                233   
2  2835781        8      2      110018              233                233   
3  2838589       11      2      110018              233                233   
4  2855214       12     21      110015              233                233   

   insuranceCompanyId  custpincodecategory  claimtype  custpincode  \
0                 131                    1          0       500001   
1                 131                    1          0       226022   
2                 228                    1          0       110018   
3                 228                    2          0       124001   
4                 131                    1          0       201301   

  memcreateddate mobilepurchasedate claimintdate  servicecntrid  \
0     2017-04-27         2016-12-02   2018-02-08           4403   
1     2017-03-06         2017-03-05   2018-02-04           4444   
2     2018-02-01         2018-01-27   2018-02-03           4388   
3     2018-01-17         2018-01-31   2018-02-04           4092   
4     2018-01-23         2018-01-20   2018-02-09           4580   

  pickupStartdate  tat_in_days  pickupStartdate_Code    claimintdate_Code  
0      2018-02-13            5   1518480000000000000  1518048000000000000  
1      2018-02-08           12   1518048000000000000  1517702400000000000  
2      2018-02-12            9   1518393600000000000  1517616000000000000  
3      2018-02-14            9   1518566400000000000  1517702400000000000  
4      2018-02-14            8   1518566400000000000  1518134400000000000

### 2. Changing selection criteria

In [659]:
feature_cols=['mobmake','hubpincode','pickuppartnerid','deliverypartnerid','insuranceCompanyId','custpincodecategory','claimtype','servicecntrid','pickupStartdate_Code','claimintdate_Code']

In [660]:
X=df[feature_cols]
X.shape

(6064, 10)

In [661]:
y=df['tat_in_days']

In [662]:
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.25,random_state=1)

### Linear Regression

In [663]:
linreg = LinearRegression()
linreg.fit(X_train,y_train)
y_pred = linreg.predict(X_test)
y_pred = y_pred.astype(int)

In [664]:
lr=np.sqrt(metrics.mean_squared_error(y_test,y_pred))
lr

4.480020142246465

### Ridge Regression

In [665]:
ridge = Ridge()
ridge.fit(X_train,y_train)
y_pred = ridge.predict(X_test)
y_pred = y_pred.astype(int)

c:\users\gunjan\appdata\local\programs\python\python37-32\lib\site-packages\sklearn\linear_model\ridge.py:147: LinAlgWarning: Ill-conditioned matrix (rcond=1.3426e-34): result may not be accurate.
  overwrite_a=True).T


In [666]:
rr=np.sqrt(metrics.mean_squared_error(y_test,y_pred))
rr

4.466010597796495

### Lasso Regression

In [667]:
model = Lasso()
model.fit(X_train,y_train)
y_pred = model.predict(X_test)
y_pred = y_pred.astype(int)

c:\users\gunjan\appdata\local\programs\python\python37-32\lib\site-packages\sklearn\linear_model\coordinate_descent.py:475: ConvergenceWarning: Objective did not converge. You might want to increase the number of iterations. Duality gap: 17336.554180940337, tolerance: 11.342131112576961
  positive)


In [668]:
lar=np.sqrt(metrics.mean_squared_error(y_test,y_pred))
lar

4.677882614172665

### Model Evaluation

In [669]:
models = pd.DataFrame({
    'Model': [ 'Linear Regression','Lasso Regression','Ridge Regression'],
    'RMSE Score': [lr,lar,rr]})
models.sort_values(by='RMSE Score', ascending=True)

Model  RMSE Score
2   Ridge Regression    4.466011
0  Linear Regression    4.480020
1   Lasso Regression    4.677883

#### RMSE Score through Ridge Regression=4.466011

### 3. Changing selection criteria

In [670]:
feature_cols=['mobmake','hubpincode','pickuppartnerid','deliverypartnerid','insuranceCompanyId','claimtype','custpincode','servicecntrid','pickupStartdate_Code','claimintdate_Code']

In [671]:
X=df[feature_cols]
X.shape


(6064, 10)

In [672]:
y=df['tat_in_days']

In [673]:
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.18,random_state=1)

### Linear Regression

In [674]:
linreg = LinearRegression()
linreg.fit(X_train,y_train)
y_pred = linreg.predict(X_test)
y_pred = y_pred.astype(int)

In [675]:
lr=np.sqrt(metrics.mean_squared_error(y_test,y_pred))
lr

4.683870695124353

### Ridge Regression

In [676]:
ridge = Ridge()
ridge.fit(X_train,y_train)
y_pred = ridge.predict(X_test)
y_pred = y_pred.astype(int)

c:\users\gunjan\appdata\local\programs\python\python37-32\lib\site-packages\sklearn\linear_model\ridge.py:147: LinAlgWarning: Ill-conditioned matrix (rcond=1.23267e-34): result may not be accurate.
  overwrite_a=True).T


In [677]:
rr=np.sqrt(metrics.mean_squared_error(y_test,y_pred))
rr

4.668595885894125

### Lasso Regression

In [678]:
model = Lasso()
model.fit(X_train,y_train)
y_pred = model.predict(X_test)
y_pred = y_pred.astype(int)

In [679]:
lar=np.sqrt(metrics.mean_squared_error(y_test,y_pred))
lar

4.680154501065767

### Model Evaluation

In [680]:
models = pd.DataFrame({
    'Model': [ 'Linear Regression','Lasso Regression','Ridge Regression'],
    'RMSE Score': [lr,lar,rr]})
models.sort_values(by='RMSE Score', ascending=True)

Model  RMSE Score
2   Ridge Regression    4.668596
1   Lasso Regression    4.680155
0  Linear Regression    4.683871

### The best model that fits to the problem is Ridge Regression with RMSE Score in selection criteria 1= 4.450771

In [681]:
X_test = dft[['mobmake','hubpincode','pickuppartnerid','deliverypartnerid','insuranceCompanyId','custpincodecategory','claimtype','custpincode','servicecntrid','pickupStartdate_Code','claimintdate_Code']]
y_pred = ridge1.predict(X_test)
y_pred = y_pred.astype(int)
dft['tat_in_days'] = y_pred
dft.to_csv('problem_2_test.csv')